File: app/src/main/java/com/lkms/data/model/java/Project.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "Project".
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Project {

    @SerializedName("projectId")
    private Integer projectId;

    @SerializedName("projectTitle")
    private String projectTitle;

    // Khóa ngoại, liên kết đến userId trong bảng User
    @SerializedName("projectLeaderId")
    private Integer projectLeaderId;
}

File: app/src/main/java/com/lkms/data/model/Project.kt
package com.lkms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Project(
    @SerialName("projectId")
    val projectId: Int? = null,

    @SerialName("projectTitle")
    val projectTitle: String? = null,

    // Khóa ngoại, liên kết đến userId trong bảng User
    @SerialName("projectLeaderId")
    val projectLeaderId: Int? = null
)